staking/transaction/transaction.go

```diff
@@ -0,0 +1,111 @@
+package transaction
+
+import (
+	"errors"
+	"io"
+	"math/big"
+	"sync/atomic"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/harmony-one/harmony/crypto/hash"
+	"github.com/harmony-one/harmony/staking/message"
+)
+
+type txdata struct {
+	message.Directive
+	Stake        interface{}
+	AccountNonce uint64   `json:"nonce"      gencodec:"required"`
+	Price        *big.Int `json:"gasPrice"   gencodec:"required"`
+	GasLimit     uint64   `json:"gas"        gencodec:"required"`
+	// Signature values
+	V *big.Int `json:"v" gencodec:"required"`
+	R *big.Int `json:"r" gencodec:"required"`
+	S *big.Int `json:"s" gencodec:"required"`
+	// This is only used when marshaling to JSON.
+	Hash *common.Hash `json:"hash" rlp:"-"`
+}
+
+// Stake is a record captuing all staking operations
+type Stake struct {
```

Comment:
This should really just by StakingTransaction. Stake reads like it's containing information about a specific stake. Not quite easy to understand.

Reply:
I want to avoid stutter though: 
```
Avoid stutter. Since client code uses the package name as a prefix when referring to the package contents, the names for those contents need not repeat the package name. The HTTP server provided by the http package is called Server, not HTTPServer. Client code refers to this type as http.Server, so there is no ambiguity. 
```

https://blog.golang.org/package-names

caller ends up being then: `transaction.StakingTransaction`  But on this one I can see it going either way, can change